application/src/test/java/org/opentripplanner/updater/GtfsRealtimeFuzzyTripMatcherTest.java

```diff
@@ -2,47 +2,146 @@
+    TRIP,
+    TEST_MODEL.stopTimesEvery5Minutes(5, TRIP, START_TIME),
+    new Deduplicator()
+  );
+  private static final TripPattern TRIP_PATTERN = TimetableRepositoryForTest
+    .tripPattern("tp1", ROUTE)
+    .withStopPattern(TimetableRepositoryForTest.stopPattern(STOP_1, STOP_2))
+    .withScheduledTimeTableBuilder(builder -> builder.addTripTimes(TRIP_TIMES))
+    .build();
+
+  @BeforeAll
+  static void setup() {
+    TRIP_TIMES.setServiceCode(SERVICE_CODE);
+    CalendarServiceData calendarServiceData = new CalendarServiceData();
+    calendarServiceData.putServiceDatesForServiceId(SERVICE_ID, List.of(SERVICE_DATE));
+    TT_REPO.addTripPattern(TRIP_PATTERN.getId(), TRIP_PATTERN);
+    TT_REPO.getServiceCodes().put(SERVICE_ID, SERVICE_CODE);
+    TT_REPO.updateCalendarServiceData(true, calendarServiceData, DataImportIssueStore.NOOP);
+    TT_REPO.index();
+  }
+
+  @Test
+  void noTripId() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().build();
+    assertEquals(TRIP.getId().getId(), matcher.match(FEED_ID, trip1).getTripId());
+  }
+
+  @Test
+  void tripIdSetButNotInSchedule() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setTripId("does-not-exist-in-schedule").build();
+    assertEquals(TRIP_ID, matcher.match(FEED_ID, trip1).getTripId());
+  }
 
   @Test
-  public void testMatch() {
-    String feedId = timetableRepository.getFeedIds().iterator().next();
+  void tripIdExistsInSchedule() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setTripId(TRIP_ID).build();
+    assertEquals(TRIP_ID, matcher.match(FEED_ID, trip1).getTripId());
+  }
+
+  @Test
+  void incorrectRoute() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setRouteId("does-not-exists").build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
+
+  @Test
+  void incorrectDateFormat() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setStartDate("ZZZ").build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
+
+  @Test
+  void incorrectDirection() {
+    var matcher = matcher();
+    TripDescriptor trip1 = matchingTripUpdate().setDirectionId(1).build();
+    assertFalse(matcher.match(FEED_ID, trip1).hasTripId());
+  }
 
-    GtfsRealtimeFuzzyTripMatcher matcher = new GtfsRealtimeFuzzyTripMatcher(
-      new DefaultTransitService(timetableRepository)
+  public static List<Function<TripDescriptor.Builder, TripDescriptor.Builder>> incompleteDataCases() {
+    return List.of(
+      TripDescriptor.Builder::clearDirectionId,
+      TripDescriptor.Builder::clearRouteId,
+      TripDescriptor.Builder::clearStartTime,
+      TripDescriptor.Builder::clearStartDate
     );
-    TripDescriptor trip1 = TripDescriptor
+  }
+
+  @MethodSource("incompleteDataCases")
+  @ParameterizedTest
+  void incompleteMatchingData(Function<TripDescriptor.Builder, TripDescriptor.Builder> modifier) {
```

Comment:
I think it might be more readable and probably wouldn't need that many more lines of code if you just wrote down the 4 test cases as individual tests instead of using this parametrized test, but I don't have that strong opinion on this.

Reply:
Done in 2be774b24293b222a92b63d5476f9d06ee2e494b